Modules/HMPID/src/HmpidTaskDigits.cxx

```diff
@@ -0,0 +1,181 @@
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+///
+/// \file       HmpidTaskDigits.cxx
+/// \author     Antonio Paz, Giacomo Volpe
+/// \brief      Class to map data from HMPID detectors
+/// \version    0.3.1
+/// \date       26/10/2021
+///
+
+//! ap  Changes:
+//     -  Histograms    hHMPIDdigitmapAvg[numCham]   back to   TH2F
+//     -  Histograms    hHMPIDchargeDist[numCham]  back to  TH1F
+
+#include <TCanvas.h>
+#include <TH1F.h>
+#include <TH2F.h>
+#include <TProfile.h>
+
+#include <TMath.h>
+#include <Framework/InputRecord.h>
+
+#include "QualityControl/QcInfoLogger.h"
+#include "HMPID/HmpidTaskDigits.h"
+#include "DataFormatsHMP/Digit.h"
+#include "DataFormatsHMP/Trigger.h"
+
+namespace o2::quality_control_modules::hmpid
+{
+
+HmpidTaskDigits::~HmpidTaskDigits()
+{
+  if (hOccupancyAvg) {
+    delete hOccupancyAvg;
+  }
+
+  for (Int_t i = 0; i < numCham; ++i) {
+    if (hHMPIDchargeDist[i]) {
+      delete hHMPIDchargeDist[i];
+    }
+    if (hHMPIDdigitmapAvg[i]) {
+      delete hHMPIDdigitmapAvg[i];
+    }
+  }
+}
+
+void HmpidTaskDigits::initialize(o2::framework::InitContext& /*ctx*/)
+{
+  ILOG(Info) << "initialize HmpidTask" << ENDM; // QcInfoLogger is used. FairMQ logs will go to there as well.
+
+  // this is how to get access to custom parameters defined in the config file at qc.tasks.<task_name>.taskParameters
+  if (auto param = mCustomParameters.find("myOwnKey"); param != mCustomParameters.end()) {
+    ILOG(Info) << "Custom parameter - myOwnKey: " << param->second << ENDM;
+  }
+
+  hOccupancyAvg = new TProfile("hOccupancyAvg", "Occupancy per DDL;;Occupancy (%)", 14, 0.5, 14.5);
+  hOccupancyAvg->Sumw2();
+  hOccupancyAvg->SetOption("P");
+  hOccupancyAvg->SetMinimum(0);
+  hOccupancyAvg->SetMarkerStyle(20);
+  hOccupancyAvg->SetMarkerColor(kBlack);
+  hOccupancyAvg->SetLineColor(kBlack);
+  for (Int_t iddl = 0; iddl < 14; iddl++)
+    hOccupancyAvg->GetXaxis()->SetBinLabel(iddl + 1, Form("%d", iddl + 1));
+  hOccupancyAvg->GetXaxis()->SetLabelSize(0.02);
+  hOccupancyAvg->SetStats(0);
+
+  getObjectsManager()->startPublishing(hOccupancyAvg);
+  getObjectsManager()->addMetadata(hOccupancyAvg->GetName(), "custom", "34");
+
+  // histos for digit charge distributions
+  for (Int_t i = 0; i < numCham; ++i) {
+
+    hHMPIDchargeDist[i] = new TH1F(Form("hHMPIDchargeDist%i", i + 1), Form("Distribution of charges in chamber %i", i + 1), 2000, 0, 2000);
+    hHMPIDchargeDist[i]->SetXTitle("Charge (ADC)");
+    hHMPIDchargeDist[i]->SetYTitle("Entries/1 ADC");
+
+    hHMPIDdigitmapAvg[i] = new TH2F(Form("hHMPIDdigitmapAvg%i", i + 1), Form("Coordinates of hits in chamber %i", i + 1), 160, 0, 160, 144, 0, 144);
+    hHMPIDdigitmapAvg[i]->SetXTitle("padX");
+    hHMPIDdigitmapAvg[i]->SetYTitle("padY");
+
+    getObjectsManager()->startPublishing(hHMPIDchargeDist[i]);
+    getObjectsManager()->addMetadata(hHMPIDchargeDist[i]->GetName(), "custom", "34");

```

Comment:
is that really needed ?

Reply:
Do you refer to lines 91 and 92? I assume yes, but you are the expert, so if they are not needed I will remove them. I thought they are needed for the histograms publishing.

Reply:
I referred to 92
It adds metadata to a published object.